internal/kafka/test/integration/kafka_create_test.go

```diff
@@ -0,0 +1,528 @@
+					deleteTestKafka(t, h, ctx, client, kafka.ID)
+				}
+			},
+		},
+		{
+			name: "should reject kafka request if there is no capacity left",
+			setup: func() {
+				// Create dummy Kafkas to fill up the cluster capacity
+				dummyKafkas := []*dbapi.KafkaRequest{}
+				for i := 1; i <= int(kasfleetshardsync.StandardCapacityInfo.MaxUnits); i++ {
+					dummyKafkas = append(dummyKafkas, mockkafkas.BuildKafkaRequest(
+						mockkafkas.WithPredefinedTestValues(),
+						mockkafkas.With(mockkafkas.NAME, fmt.Sprintf("dummy-kafka-%d", i)),
+						mockkafkas.With(mockkafkas.CLUSTER_ID, clusterID),
+						mockkafkas.With(mockkafkas.CLOUD_PROVIDER, kafkaRequestPayload.CloudProvider),
+						mockkafkas.With(mockkafkas.REGION, kafkaRequestPayload.Region),
+						mockkafkas.With(mockkafkas.OWNER, account2.Username()),
+						mockkafkas.With(mockkafkas.ORGANISATION_ID, account2.Organization().ExternalID()),
+						mockkafkas.With(mockkafkas.INSTANCE_TYPE, types.STANDARD.String()),
+						mockkafkas.With(mockkafkas.SIZE_ID, "x1"),
+					))
+				}
+
+				db := h.DBFactory().New()
+				err := db.Create(&dummyKafkas).Error
+				g.Expect(err).NotTo(gomega.HaveOccurred(), "failed to create dummy kafkas")
+			},
+			assert: func() {
+				_, resp, err := common.WaitForKafkaCreateToBeAccepted(ctx, kafkatest.TestServices.DBFactory, client, kafkaRequestPayload)
+				if resp != nil {
+					resp.Body.Close()
+				}
+				g.Expect(err).NotTo(gomega.HaveOccurred(), "Error posting object:  %v", err)
+				// get a 403 status code as there is no capacity left
+				g.Expect(resp.StatusCode).To(gomega.Equal(http.StatusForbidden), "kafka should have been rejected as no capacity is left")
+			},
+			cleanup: func() {
+				// delete dummy kafkas to free up resources on the OSD cluster
+				db := h.DBFactory().DB.Model(&dbapi.KafkaRequest{})
+				var kafkas []*dbapi.KafkaRequest
+				err := db.Scan(&kafkas).Error
+				g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to get all kafka in the database")
+
+				ctx := h.NewAuthenticatedContext(account2, nil)
+
+				for _, kafka := range kafkas {
+					deleteTestKafka(t, h, ctx, client, kafka.ID)
+				}
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		testcase := tc
+		t.Run(testcase.name, func(t *testing.T) {
+			if testcase.setup != nil {
+				testcase.setup()
+			}
+
+			testcase.assert()
+
+			if testcase.cleanup != nil {
+				testcase.cleanup()
+			}
+		})
+	}
+}
+
+func TestKafkaCreate_ValidatePlanParam(t *testing.T) {
+	g := gomega.NewWithT(t)
+
+	ocmServer := mocks.NewMockConfigurableServerBuilder().Build()
+	defer ocmServer.Close()
+
+	h, client, teardown := kafkatest.NewKafkaHelperWithHooks(t, ocmServer, nil)
+	defer teardown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	clusterID, getClusterErr := common.GetRunningOsdClusterID(h, t)
```

Comment:
we could instead directly seed the cluster in the database with a `ready` status to save some tests time